build.gradle

```diff
@@ -1345,6 +1349,62 @@ project(':raft') {
   }
 }
 
+project(':server-common') {
+  archivesBaseName = "kafka-server-common"
```

Comment:
Do we want to move the classes in storage-api here? It helps reduce one module.

Reply:
I plan to do that in another PR. I have not yet decided whether to have a separate sub module like `server-api` or part of `server-common` module.

Reply:
This is a good question and worth thinking about a bit more. I agree we can do it in a separate PR. I think the main thing to consider is how we differentiate between public classes exposed to users and common classes to be used by the kafka server modules.

Reply:
Created https://issues.apache.org/jira/browse/KAFKA-12757 to discuss more on this.